akka-stream/src/main/scala/akka/stream/impl/fusing/GraphInterpreter.scala

```diff
@@ -98,12 +102,12 @@ private[stream] object GraphInterpreter {
 
       for (i ← 0 until connectionCount) {
```

Comment:
I will phase out all of these (I have experienced the "performance" of ranges before, also in the other interpreter). What was important here is that the design is such that it only needs a few linear passes over arrays, and being able to reuse most of the arrays without the need for copy (like inOwners, outOwners).